### PR TITLE
[d3-array] Update types to v2.3

### PR DESCRIPTION
**Pull request by Methuselah96**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/d3/d3-array/releases/tag/v2.3.0
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

Related: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/38939

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## d3-array ([<kbd>unpkg</kbd>](https://unpkg.com/browse/d3-array@latest/))
was missing the following properties:
1. scan
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 333fb294d908a8939c1edd19db8a46c5c3d83582
</p>

File: types/d3-array/d3-array-tests.ts [hunk 1]
```diff
@@ -341,99 +347,133 @@ stringyNumbersArray = ['0', '2', '3', '4', '7', '8'];
 dateArray = [new Date(2010, 1, 1), new Date(2011, 1, 1), new Date(2012, 1, 1), new Date(2013, 1, 1)];
 typedArray = Uint8Array.from(numbersArray);
 
-// scan() ----------------------------------------------------------------------
+// least() ----------------------------------------------------------------------
```
Comment (Ledragon) [inline types/d3-array/d3-array-tests.ts:350]: was scan fully removed? I guess it makes sense to remove the tests as it is deprectaed though

**Review by Ledragon — CHANGES_REQUESTED** [review]
Unless I miss something, there are no test for the updated version of `quantile` nor `quantileSorted`, I think these should be added before merge

File: types/d3-array/d3-array-tests.ts (hunk 1 above)
Comment (Methuselah96) [reply types/d3-array/d3-array-tests.ts:350]: The documentation for it was removed in https://github.com/d3/d3-array/commit/09b8ff21742b365a7090e728120640fb45464637 (before the `2.3.0` release), so I assumed it was just an oversight that it wasn't fully removed in `2.0.0`.

**Comment by Methuselah96** [comment]
@Ledragon The tests for `quantileSorted` are right [here](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/48733/files#diff-538ca513c7ce4d105b7b78db1bae532e052c076c6b81e899cb605583aecec395R290-R295) and the tests for `quantile` are right above that.

**Comment by Ledragon** [comment]
Sorry I'm blind...

**Review by Ledragon — APPROVED** [review]

**Pull request merged**